docs/rules/no-context.md

```diff
@@ -0,0 +1,51 @@
+# Prevent usage of context (no-context)
+
+[Context](https://facebook.github.io/react/docs/context.html) is an advanced and experimental feature. The API is likely to change in future releases. Most applications will never need to use context. Especially if you are just getting started with React, you likely do not want to use context. Using context will make your code harder to understand because it makes the data flow less clear. It is similar to using global variables to pass state through your application.
+
+## Rule Details
+
+The following patterns are considered warnings:
+
+```js
+class Button extends React.Component {
+  render() {
+    return (
+      <button style={{background: this.context.color}}>
+        {this.props.children}
+      </button>
+    );
+  }
+}
+Button.contextTypes = {
+  color: React.PropTypes.string
+};
+
+const Button = ({children}, context) =>
```

Comment:
Also, it might be nice to change `{children}` to `props` to simplify the example, which might make it clearer to folks who haven't encountered destructuring like this before.